tests/queries.py

```diff
@@ -285,19 +287,18 @@ def print_layer_stat(layers):
 
 
 # TODO this can be a util function
-def get_latest_layer(deployment):
+def get_latest_layer(deployment, num_miners):
     layers = get_layers(deployment)
-    layers.sort(reverse=True)
-    if len(layers) == 0:
-        return 0
-    return layers[0]
+    for layer, node_cnt in sorted(layers.items(), key=lambda t: -t[0]):
+        if node_cnt >= num_miners:
+            return layer
 
```

Comment:
this is not necessary to iterate on if you will return the max layer as in the last comment

Reply:
See reply to prev. comment.